File: src/code/ts/euler-phi.ts
//#region main
export function eulerPhi(n: bigint) {
  let res: bigint = n;
  for (let i = 2n; i * i <= n; i++) {
    if (n % i === 0n) {
      res = (res / i) * (i - 1n);
      while (n % i === 0n) n /= i;
    }
  }
  if (n > 1n) res = (res / n) * (n - 1n);
  return res;
}

//#endregion main

File: src/code/py/euler_phi.py
import math

#region main
def euler_phi(n: int) -> int:
    res = n
    for i in range(2, math.floor(math.sqrt(n)) + 1):
        if n % i == 0:
            res = (res // i) * (i - 1)
            while n % i == 0:
                n //= i
    if res > 1:
        res = (res // n) * (n - 1)
    return res
#endregion main